### PR TITLE
Help to complete referral

### PR DESCRIPTION
**Pull request by ian-leggett**
## Description of change

If a user cannot complete a referral we need a page which offers further guidance on how they could possibly action the referral.

## Test instructions

You will need to post a referral to the back end. Method POST `/v4/company-referral/{id}`

For more info see the API docs - https://api.datahub.dev.uktrade.io/docs

Once you have a referral posted just access the referral via /referral/{id}/help

## Screenshots
Screenshot 2020-02-26 at 14 54 36

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/apps/referrals/apps/help/client/ReferralHelp.jsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      </>
+    )}
+  </>
+)
+
+export default connect(state2props)(({ id, company, sendingAdviser }) => {
+  // Although this input creates impurity I think for simplicity its ok in this context
+  const host = `${location.protocol + location.host}`
+  const relReferralLink = `/referrals/${id}`
+  const absReferralLink = `${host}/referrals/${id}`
```
Comment (rafenden) [inline src/apps/referrals/apps/help/client/ReferralHelp.jsx]: Why not to use the urls module here? Does this link have to be absolute?

Comment (ian-leggett) [reply src/apps/referrals/apps/help/client/ReferralHelp.jsx]: Good point, I have added the urls module 👍 The absolute part is only for content purposes as for this first iteration the user just needs to be able select the whole link for copy and paste.

File: src/apps/referrals/apps/help/client/ReferralHelp.jsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      </>
+    )}
+  </>
+)
+
+export default connect(state2props)(({ id, company, sendingAdviser }) => {
+  // Although this input creates impurity I think for simplicity its ok in this context
+  const host = `${location.protocol}${location.host}`
+  const relReferralLink = urls.referrals.details(id)
+  const absReferralLink = `${host}${urls.referrals.details(id)}`
```
Comment (rafenden) [inline src/apps/referrals/apps/help/client/ReferralHelp.jsx]: You are missing two forward slashes, i.e. `//`

Comment (ian-leggett) [reply src/apps/referrals/apps/help/client/ReferralHelp.jsx]: Good spot 👍

File: test/functional/cypress/specs/referrals/referral-help-spec.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        'have.text',
+        'Help to complete this Referral'
+      )
+    })
+
+    it('should render the content and conform to design', () => {
+      cy.get('#referral-help h2:first-child')
+        .should('have.text', 'More information needed')
+        .next()
+        .should('have.prop', 'tagName', 'P')
```
Comment (rafenden) [inline test/functional/cypress/specs/referrals/referral-help-spec.js]: There is a simpler way:
```suggestion
        .should('match, 'p')
```

**Comment by ian-leggett** [comment]
> Should the breadcrumbs not be Companies > companyName > Complete referral > Help to complete ?

It is like that in the design however after speaking with Billy and Nitesh it shouldn't be nested in companies, referrals is its own thing like company lists.

**Comment by ian-leggett** [comment]
> Should the breadcrumbs not be Companies > companyName > Complete referral > Help to complete ?

It is like that in the design however after speaking with Billy and Nitesh it shouldn't be nested in companies, referrals is its own thing like company lists.

File: src/apps/referrals/apps/help/client/ReferralHelp.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    >
+      {() =>
+        company && (
+          <>
+            <H2 size={LEVEL_SIZE[3]}>More information needed</H2>
+            <p>
+              If you need additional details or information about the referral,
+              contact the adviser who sent the referral by email:{' '}
+              <AdviserDetails {...sendingAdviser} />, Or find more contact
+              details on Digital Workspace
```
Comment (mforner13) [inline src/apps/referrals/apps/help/client/ReferralHelp.jsx]: Or should be in lowercase and it needs a full stop after Workspace :)

Comment (ian-leggett) [reply src/apps/referrals/apps/help/client/ReferralHelp.jsx]: Ah yes and it needs to link to Digital workspace. Thanks

**Pull request merged**